kb/2023-08-11/using_netapp_third_party_storage.md

```diff
@@ -0,0 +1,154 @@
+Longhorn storage; it will not replace Longhorn.
+Virtual machine images will still be stored using Longhorn.
+
+This has been tested with Harvester 1.2.0 and Trident v23.07.0.
+
+This procedure only works to access storage via iSCSI, not NFS.
+
+Note also that 3rd party storage classes (including those based on Trident)
+can only be used for non-boot volumes of Harvester VMs.
+
+== Detailed Instructions
+
+We are assuming that before beginning this procedure,
+a Harvester cluster and a NetApp ONTAP storage system
+are both installed and configured for use.
+
+Follow all of the following steps, but ONLY on the initial cluster node.
+
+The last step (setting up multipathd) should be done on all nodes,
+AFTER the initial node has had Trident installed.
+
+1. Read the NetApp Astra Trident documentation
+       https://docs.netapp.com/us-en/trident/
+       https://docs.netapp.com/us-en/trident/trident-get-started/kubernetes-deploy-operator.html
+   We are describing the "manual deployment" procedure here.
+2. Log into the Harvester cluster in one of the nodes.
+3. Download and extract the Trident software into /tmp on the Harvester node.
+   Note that the Harvester node will not have "wget" but it does have
+   "curl", so the command to download the software is adjusted accordingly:
+   - `cd /tmp`
+   - `curl -L -o trident-installer-23.07.0.tar.gz https://github.com/NetApp/trident/releases/download/v23.07.0/trident-installer-23.07.0.tar.gz`
+   - `tar -xf trident-installer-23.07.0.tar.gz`
+   - `cd trident-installer`
+4. Prepare, install Trident, and verify the installation
+   This part is a summary of the relevant steps from the NetApp Trident
+   website.
+4.a. Create the TridentOrchestrator CRD
+   - `kubectl create -f deploy/crds/trident.netapp.io_tridentorchestrators_crd_post1.16.yaml`
+4.b. Deploy the Trident operator (assuming the "trident" namespace)
+   - `kubectl apply -f deploy/namespace.yaml`
+   - `kubectl apply -f deploy/bundle_post_1_25.yaml`
+   The operator, deployment, and replicasets can be verified with:
+   - `kubectl get all -n trident`
+4.c. Create the TridentOrchestrator and install Trident
+   - `kubectl create -f deploy/crds/tridentorchestrator_cr.yaml`
+   This will take a while (a few minutes); you can check progress with
+   - `kubectl desribe torc trident`
+   The last few lines of output will tell you whether the installation
+   has completed successfully.  When it has finished, the last few
+   lines should look something like this:
+
+```
+Events:
+    Type Reason Age From Message ---- ------ ---- ---- -------Normal
+    Installing 74s trident-operator.netapp.io Installing Trident Normal
+    Installed 67s trident-operator.netapp.io Trident installed
+```
+
+4.d. Verify the installation
+   - `kubectl get pods -n trident`
+   - `./tridentctl -n trident version`
+
+5. Install a backend.
+   This part is specific to Harvester.
+5.a.    Put the following into a text file, for example /tmp/backend.yaml
```

Comment:
we should may be add some explanation for the fields in the backend.yaml? or also point to the official spec?

Reply:
I didn't think of this.  I will have to look up the right doc.  At the moment, this part is magic to me.